policy-controller/k8s/index/src/lib.rs

```diff
@@ -63,6 +63,9 @@ pub struct ClusterInfo {
 
     /// The cluster-wide default protocol detection timeout.
     pub default_detect_timeout: time::Duration,
+
+    /// The networks that probes are expected to be from.
+    pub probe_networks: Option<Vec<IpNet>>,
```

Comment:
AFAICT, these are always converted from `IpNet`s to `NetworkMatch`es when used in the indexer, by iterating over this and converting them using `Into`. Would it make more sense to just store this as a 
```suggestion
    pub probe_networks: Option<Vec<NetworkMatch>>,
```
which can then just be `.clone()`ed into each `InboundServer`? that might be more efficient, as the clone could probably just be one big `memcpy`, rather than iterating over and mutating a new allocation. also, i think this would make the code simpler. :)

Reply:
I'm not sure whether this makes sense (I'd need to refresh myself on NetworkMatch), but I don't think we should be trying to optimize around allocations here.